File: bungee/src/main/java/live/turna/phenyl/bungee/listener/bungee/BungeeOnLoginEvent.java
package live.turna.phenyl.bungee.listener.bungee;

import live.turna.phenyl.bungee.BungeePhenyl;
import live.turna.phenyl.common.eventhandler.server.OnLoginEvent;
import net.md_5.bungee.api.event.ServerConnectedEvent;
import net.md_5.bungee.api.plugin.Listener;
import net.md_5.bungee.event.EventHandler;

/**
 * <b>BungeeOnLoginEvent</b><br>
 * Called when a server is ready to take control of a player instance.
 * Send a message to QQ group if {@code on_join} is not set to "disabled".<br>
 * Send a message to all online players if {@code on_join_broadcast} is not set to "disabled".
 *
 * @since 2021/12/4 21:52
 */
public class BungeeOnLoginEvent extends OnLoginEvent<BungeePhenyl> implements Listener {

    public BungeeOnLoginEvent(BungeePhenyl plugin) {
        super(plugin);
    }

    @EventHandler
    public void onLoginEvent(ServerConnectedEvent e) {
        super.fill(phenyl.getSenderFactory().wrap(e.getPlayer()), messageUtils.getServerName(e.getServer().getInfo().getName()));
        super.handle();
    }
}

File: bungee/src/main/java/live/turna/phenyl/bungee/listener/bungee/BungeeOnLoginEvent.kt
package live.turna.phenyl.bungee.listener.bungee

import live.turna.phenyl.bungee.BungeePhenyl
import live.turna.phenyl.common.eventhandler.server.OnLoginEvent
import net.md_5.bungee.api.event.ServerConnectedEvent
import net.md_5.bungee.api.plugin.Listener
import net.md_5.bungee.event.EventHandler

class BungeeOnLoginEvent(plugin: BungeePhenyl) : OnLoginEvent<BungeePhenyl>(plugin), Listener {
    @EventHandler
    fun onLoginEvent(e: ServerConnectedEvent) {
        super.fill(phenyl.senderFactory.wrap(e.player), messageUtils.getServerName(e.server.info.name))
        super.handle()
    }
}